proxy/proxy.go

```diff
@@ -421,8 +424,8 @@ func isProtocolSupported(request *http.Request) bool {
 	return request.ProtoMajor == 1 && (request.ProtoMinor == 0 || request.ProtoMinor == 1)
 }
 
-func isLoadBalancerHeartbeat(request *http.Request) bool {
-	return request.UserAgent() == "HTTP-Monitor/1.1"
+func isLoadBalancerHeartbeat(request *http.Request, healthCheckUserAgent string) bool {
```

Comment:
This could be a private method of proxy struct. This way you can consume the `p.healthCheckUserAgent` directly, and you decouple the implementation details of `isLoadBalancerHeartbeat()` from the caller.

But that is a minor detail.